### DISCUSSION TITLE
Error as3mxml.java.path in settings does not point to a valid executable on Mac M1

### DISCUSSION THREAD

**Discussion by smival** [BowlerHatLLC/vscode-as3mxml#610] [Q&A] [answered]
<img width="226" alt="image" src="https://user-images.githubusercontent.com/2552793/176770978-7056e16c-fd34-46c0-8030-d9d7cb087b4a.png">

 "It cannot be a directory" - the question is that exactly execution it needs?
Sorry I am not a Java expert, usually any SDKs needs correct PATH only,
perhaps mac silicon has some different adjustment?

java -version:
openjdk version "18.0.1.1" 2022-04-22
OpenJDK Runtime Environment Homebrew (build 18.0.1.1+0)
OpenJDK 64-Bit Server VM Homebrew (build 18.0.1.1+0, mixed mode, sharing)

echo $PATH:
/opt/homebrew/opt/openjdk/bin:/opt/homebrew/opt/node@14/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands

Thanks for any help

**Comment by joshtynjala** [comment, marked as answer]
Yes, the AS3 & MXML extension for VSCode is capable of checking the system PATH environment variable to find the JDK. However, I don't believe that I've ever tested it with Java installed with Homebrew. I've always used an official _.pkg_ installer. Previously, from Oracle. More recently, from one of the OpenJDK community versions, such as Adoptium.

Regardless, if you've set `as3mxml.java.path` in VSCode's settings, it will use that value instead of the PATH value. The error message that you mentioned indicates that you've set `as3mxml.java.path` in VSCode's settings, so it will ignore the PATH. What value did you use? Based on your PATH environment variable, I suspect that the correct value should be "/opt/homebrew/opt/openjdk/bin/java". By the way, you can run `which java` in a terminal to see the path to the `java` executable that was detected from your PATH. That's the path you can use for `as3mxml.java.path`.

> perhaps mac silicon has some different adjustment?

My primary machine is an Apple Silicon Mac Mini. I did not need to make any changes to detect Java when I switched to this new machine. Everything in AS3 & MXML for VSCode just worked.

**Reply by smival** [reply]
thank you for detailed reply, yes actually OpenJDK works fine with "as3mxml.java.path": "/opt/homebrew/opt/openjdk/bin/java"
